lib/url_to_subscriber_list_criteria.rb

```diff
@@ -0,0 +1,157 @@
+require 'uri'
+
+class UrlToSubscriberListCriteria
+  MISSING_LOOKUP = "*** MISSING KEY ***"
+  EMAIL_SUPERTYPE = "email_document_supertype"
+  GOVERNMENT_SUPERTYPE = "government_document_supertype"
+  class UnprocessableUrl < StandardError; end
+
+  def initialize(url, static_data = StaticData)
+    @url = URI.parse(url.strip)
+    @static_data = static_data
+    @missing_lookups = []
+  end
+
+  def convert
+    @convert ||= begin
+      hash = map_url_to_hash.dup
+      if hash["links"]
+        links = hash["links"].each_with_object({}) do |(key, values), result|
+          result[key] = values.map { |value| lookup_content_id(key, value) }
+        end
+        hash.merge!("links" => links)
+      end
+      hash
+    end
+  end
+
+  def missing_lookup
+    convert
+    @missing_lookups.to_sentence
```

Comment:
This needs ` if @missing_lookups.present?` to stop it always thinking there are missing lookups.